### ISSUE TITLE
Corner-case issue with single item enums

### ISSUE THREAD

**Issue by jnd77** [jqno/equalsverifier#125]
I came across the following issue, when using an enum with a single item.
The test (see piece of code below) is failing, but is passing when I change the enum to {OK, NOT_OK}.
It's very small issue, but thought it was worth sharing. :)

``` java
public class EnumIssue {

    public enum Status {OK};

    private final Status status;

    public EnumIssue() {
        status = Status.OK;
    }

    @Override
    public int hashCode() {
        final int prime = 31;
        int result = 1;
        result = prime * result + status.hashCode();
        return result;
    }

    @Override
    public boolean equals(Object obj) {
        if (this == obj)
            return true;
        if (obj == null)
            return false;
        if (getClass() != obj.getClass())
            return false;
        EnumIssue other = (EnumIssue) obj;
        if (status != other.status)
            return false;
        return true;
    }
}
```

And the failing test is:

``` java
public class EnumIssueTest {

    @Test
    public void equals() {
        EqualsVerifier.forClass(EnumIssue.class)
            .usingGetClass()
            .suppress(Warning.NULL_FIELDS)
            .verify();
    }
}

```

**Comment by jqno** [comment]
Indeed, thanks for catching this. I'll fix it in the next release.

**Comment by jqno** [comment]
I just released version 1.7.5, which fixes this.

**Comment by jelford** [comment]
This failure comes back with `allFieldsShouldBeUsed`:

```
@Test
    public void equals() {
        EqualsVerifier.forClass(EnumIssue.class)
            .usingGetClass()
            .suppress(Warning.NULL_FIELDS)
            .allFieldsShouldBeUsed()                    // <-- this
            .verify();
    }
```

I can open a new issue if you'd prefer (might submit a PR later if you agree this shouldn't fail).

**Comment by jqno** [comment]
Which version of EqualsVerifier are you using? `allFieldsShouldBeUsed()` was removed in version 2.0.

**Comment by jelford** [comment]
Ah, my bad; we're on 1.7.8. Sorry!

**Comment by jqno** [comment]
OK, I was able to reproduce this in 1.7.8.

In version 2.0 and up, `allFieldsShouldBeUsed` is the default behaviour, but there, this issue doesn't come up. Is there a reason why you can't upgrade to 2.0? Or does it need to be fixed in the 1.7.x as well?

**Comment by jelford** [comment]
Sorry I should have been clearer in my previous; we will look to upgrade to
the latest version. If that's not possible for some reason, I'll send a
patch rather than worrying you with maintaining an old version. Thanks for
looking at this.

**Comment by jqno** [comment]
Hi,
It's not a big hassle to fix this in the older version (I can simply port the fix I have in 2.0). But if you intend to upgrade anyway, that would be a lot easier indeed :).
Jan
